awx/ui_next/src/components/FormField/FormSubmitError.jsx

```diff
@@ -2,62 +2,21 @@ import React, { useState, useEffect } from 'react';
 import { useFormikContext } from 'formik';
 import { Alert } from '@patternfly/react-core';
 import { FormFullWidthLayout } from '../FormLayout';
-
-const findErrorStrings = (obj, messages = []) => {
-  if (typeof obj === 'string') {
-    messages.push(obj);
-  } else if (typeof obj === 'object') {
-    Object.keys(obj).forEach(key => {
-      const value = obj[key];
-      if (typeof value === 'string') {
-        messages.push(value);
-      } else if (Array.isArray(value)) {
-        value.forEach(arrValue => {
-          messages = findErrorStrings(arrValue, messages);
-        });
-      } else if (typeof value === 'object') {
-        messages = findErrorStrings(value, messages);
-      }
-    });
-  }
-  return messages;
-};
+import sortErrorMessages from './sortErrorMessages';
 
 function FormSubmitError({ error }) {
   const [errorMessage, setErrorMessage] = useState(null);
-  const { setErrors } = useFormikContext();
+  const { values, setErrors } = useFormikContext();
 
   useEffect(() => {
-    if (!error) {
-      return;
+    const { formError, fieldErrors } = sortErrorMessages(error, values);
```

Comment:
:+1: nice. I like that we're pulling this into a separate module/function for testing.